### DISCUSSION TITLE
Having troubles using static info in attributes

### DISCUSSION THREAD

**Discussion by Ramaddan** [davidrapan/ha-solarman#226] [Q&A]
Hi,

I was hoping for the possibility to see the following information to be returned:

- communication board version number
- control board version number
- inverter id

This information was very useful to easily recognize the differences between the different inverters if more than one are displayed in the app.

It also made it easy to keep track of firmware versions in case they needed to be changed or updated.

This extra information is also useful in case a person wants to receive automated notifications of older firmware when an additional inverter is added or an older one replaced.

Thanks again for the great work

**Comment by davidrapan** [comment]
Try look into attributes of sensor named `Device`.

**Comment by Ramaddan** [comment]
Hi,

I already looked everywhere and could not find anything

Thanks

**Comment by davidrapan** [comment]
What profile?

**Comment by Ramaddan** [comment]
Hi,

Deye Hybrid

I also noticed that if the generator was turned ON automatically by the inverter due to low battery, there is no sensor to see status

So a person would also not know that the generator turned ON

There is only a sensor for if the generator was turned ON manually with Gen force

It means a labelling difference should also be made between Gen Force and automatically turned on Generator

And if a sensor can also be added to show status in general, for whether it is ON due to Gen force or automatic

So I guess that makes it three sensors, such as:

Gen Force (can toggle)
Gen Auto (just display)
Gen Status (just display)

Thanks

**Comment by davidrapan** [comment]
Your missing info is under this button:
![device_attributes](https://github.com/user-attachments/assets/450f7af0-e098-4a2d-8335-1930cdfb502a)

And the entities are sufficiently differentiated by their type. As switch which you can control thus turn generator on/off and binary_sensor which indicates if its on or not. It really can't be understood in any other way than that. 😉

**Comment by Ramaddan** [comment]
Yes, I see it in attributes under the following sensor

sensor.yourinverter_inverter_device

However, how do you access them directly, if I want to display or use them?

As for the generator status...

binary_sensor.yourinverter_inverter_generator

I do use that sensor, and it works fine when the generator is turned ON manually

But when it turns on automatically, nothing shows

Thank you for your help

**Comment by davidrapan** [comment]
> However, how do you access them directly, if I want to display or use them?

https://community.home-assistant.io/t/display-an-entitys-attribute/197721/4

> I do use that sensor, and it works fine when the generator is turned ON manually.

That's rather about how the device itself treats this registry value, integration just shows it. So I really don't know why your device does not signal automatically turned on generator.

I was looking into the docs but could not find any other register for that, sadly...

**Comment by Ramaddan** [comment]
> https://community.home-assistant.io/t/display-an-entitys-attribute/197721/4

Ok, trying to do it, but nothing displaying so far

> That's rather about how the device itself treats this registry value, integration just shows it. So I really don't know why your device does not signal automatically turned on generator.

Then, It might be using logic as follows:

If "switch.yourinverter_inverter_battery_generator_charging" is set to ON

AND

"sensor.yourinverter_inverter_battery" is less than "number.yourinverter_inverter_battery_generator_charging_start"

Then turn ON generator

However, it means there still should be some register somewhere to do the actual turning ON in such a case

But I think we can still use the above logic to replicate the scenario, but unfortunately nothing physical to test in case something goes wrong

**Comment by davidrapan** [comment]
> However, it means there still should be some register somewhere to do the actual turning ON in such a case

In SG04LP3 for example there is a register which shows state of the AC relay from which is read for example if grid is on but also even when the generator/smartload is on.

But could not find this register for Hybrid though...

**Comment by Ramaddan** [comment]
Weird, because I thought this was there before

I did not have this problem in the past

Also, I am still unable to display the version number in the dashboard

This is my yaml

```
type: entities
entities:
  - entity: sensor.yourinverter_inverter_device
    type: attribute
    name: control_board_firmware_version
    attribute: control_board_firmware_version

```

**Comment by davidrapan** [comment]
Because static info does not need their own sensor.

You have to set the attribute to the exact name as it is when you list the attributes so it's `Control Board Firmware Version` presumably.

**Comment by Ramaddan** [comment]
Sorry, that was my latest attempt

I tried what you suggested first

**Comment by davidrapan** [comment]
It works flawlesly for me:
![dashboard_inverter_device_attribute](https://github.com/user-attachments/assets/def2d67a-7cf2-4a4f-9f05-f5454890847a)

`yaml` looks like this:
```
type: entity
entity: sensor.inverter_device
attribute: Control Board Firmware Version
```
